app/controllers/tournaments_controller.rb

```diff
@@ -1,16 +1,37 @@
+require 'json'
+
 class TournamentsController < ApplicationController
 
+  include TournamentsHelper
```

Comment:
I think this should be a class, not a module
When it's a module, when you call `success_per_question` it's less readable - it's not clear where this code is. If several modules are included, it could be in either one of them.
It's much more readable to have it in a class, called say `StatisticsCalculator`, and have it as a class function.
Then you can just call:
`StatisticsCalculator.success_per_question`
And it's clear where the code is

Reply:
Also if the class name is specific, that is `StatisticsCalculator` vs. `TournamentsHelper`, it helps separation of concerns - that class will only hold statistics calculations, not general unrelated tournament functions

Reply:
Agree, Where should this class be added in the project tree?

Reply:
In app/services